src/data/Data.js

```diff
@@ -27,12 +27,12 @@ Data.prototype = {
 			// configurability must be true to allow the property to be
 			// deleted with the delete operator
 			} else {
-				descriptor[ this.expando ] = {
+				descriptor = {
 					value: value,
 					writable: true,
 					configurable: true
 				};
-				Object.defineProperties( owner, descriptor );
+				Object.defineProperty( owner, this.expando, descriptor );
```

Comment:
This whole block can be simpler, and doesn't even need a `try`; see 95fb798980d7e404c413e29e20016db9052e2bf2.